nixos/modules/services/backup/burp.nix

```diff
@@ -0,0 +1,451 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  libDir = "/var/lib/burp";
+
+  cfg = config.services.burp;
+  burp_conf = pkgs.writeText "burp.conf" ''
+    mode = client
+    pidfile = /run/burp/burp.pid
+
+    port = ${toString cfg.client.port}
+    status_port = ${toString cfg.client.status_port}
+    server = ${cfg.client.server}
+    password = ${cfg.client.password}
+    cname = ${cfg.client.cname}
+
+    ca_burp_ca = ${pkgs.burp}/bin/burp_ca
+    ca_csr_dir = ${cfg.client.ca_csr_dir}
+    ssl_cert_ca = ${cfg.client.ssl_cert_ca}
+    ssl_cert = ${cfg.client.ssl_cert}
+    ssl_key = ${cfg.client.ssl_key}
+    ssl_key_password = ${cfg.client.ssl_key_password}
+    ssl_peer_cn = ${cfg.client.ssl_peer_cn}
+
+    ${concatMapStringsSep "\n" (x: "include = " + x) cfg.client.includes}
+
+    ${concatMapStringsSep "\n" (x: "exclude = " + x) cfg.client.excludes}
+
+    nobackup = ${cfg.client.nobackup}
+
+    ${cfg.client.extraConfig}
+  '';
+
+  burp_server_conf = pkgs.writeText "burp-server.conf" ''
+    mode = server
+    pidfile = /run/burp/burp.server.pid
+
+    user = ${cfg.server.user}
+    group = ${cfg.server.group}
+
+    clientconfdir = ${cfg.server.clientconfdir}
+    ${concatMapStringsSep "\n" (x: "address = " + x) cfg.server.addresses}
+    port = ${toString cfg.server.port}
+    status_port = ${toString cfg.server.status_port}
+    directory = ${cfg.server.directory}
+    umask = ${toString cfg.server.umask}
+
+    ca_conf = ${cfg.server.ca_conf}
+    ca_name = ${cfg.server.ca_name}
+    ca_server_name = ${cfg.server.ca_server_name}
+    ca_burp_ca = ${pkgs.burp}/bin/burp_ca
+
+    ssl_cert_ca = ${cfg.server.ssl_cert_ca}
+    ssl_cert = ${cfg.server.ssl_cert}
+    ssl_key = ${cfg.server.ssl_key}
+    ssl_dhfile = ${cfg.server.ssl_dhfile}
+    ssl_key_password = ${cfg.server.ssl_key_password}
+    ssl_peer_cn = ${cfg.server.ssl_peer_cn}
+
+    ${concatMapStringsSep "\n" (x: "keep = " + toString x) cfg.server.keeps}
+
+    timer_script = ${cfg.server.timer_script}
+    ${concatMapStringsSep "\n" (x: "timer_arg = " + x) cfg.server.timer_args}
+
+    ${cfg.server.extraConfig}
+  '';
+
+in {
+  options = {
+    services.burp.client = {
+      enable = mkEnableOption "BURP client";
+
+      frequency = mkOption {
+        type = types.str;
```

Comment:
Should this be an enum?

Reply:
This translates to [systemd OnCalendar time specifications](https://www.freedesktop.org/software/systemd/man/systemd.time.html]. A enum maybe restrict more than desired.